core/src/apps/homescreen/device_menu.py

```diff
@@ -77,7 +87,9 @@ async def handle_device_menu() -> None:
         connected_idx = _find_device(connected_addr, bonds)
         if __debug__:
             log.debug(__name__, "connected: %s (%s)", connected_addr, connected_idx)
-        hostname_map = {e.mac_addr: e.host_name for e in paired_cache.load()}
+        hostname_map = {
+            e.mac_addr: (e.host_name, e.app_name) for e in paired_cache.load()
```

Comment:
Since we need both `host_name` and `app_name`, it would be probably simpler to keep the following map:

```python
entry_map = {e.mac_addr: e for e in paired_cache.load()}
```